### PR TITLE
Mspsds 193 add businesses

### PR DESCRIPTION
**Pull request by AnthonyPengelly**

### CODE REVIEW

File: app/assets/javascripts/investigations.js [hunk 1]
```diff
@@ -1,10 +1,12 @@
-// Place all the behaviors and hooks related to the matching controller here.
-// All this logic will automatically be available in application.js.
-// You can use CoffeeScript in this file: http://coffeescript.org/
 $(document).on("turbolinks:load", function() {
-    $(".js-investigation-products").select2({
+    addSelect2AjaxSearchToElement($(".js-investigation-products"), "/products", mapAjaxDataToProduct);
+    addSelect2AjaxSearchToElement($(".js-investigation-businesses"), "/businesses", mapAjaxDataToBusiness);
```
[the remark this answers is not in the document]
Comment (AnthonyPengelly) [reply app/assets/javascripts/investigations.js]: I will move the addSelectAjaxSearchToElement into search, this line is related to businesses, but it is on the investigations page, which is kind of how I see the breakup of the JS files atm.

File: docs/Deploy from scratch.md [hunk 2]
```diff
@@ -59,6 +59,8 @@ Create the app using the current repository
 
     # Add API key created in Notify
     cf set-env mspsds-int NOTIFY_API_KEY XXX
+    # Add API key created in Companies house
+    cf set-env mspsds-int COMPANIES_HOUSE_API+KEY XXX
```
[the remark this answers is not in the document]
Comment (AnthonyPengelly) [reply docs/Deploy from scratch.md]: Nice spot!

File: app/assets/javascripts/investigations.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (broder) [reply app/assets/javascripts/investigations.js]: Of course! Sorry for missing the `investigation` bit in the class name 😔

File: app/helpers/businesses_helper.rb [hunk 3]
```diff
@@ -1,62 +1,57 @@
...
-  def companies_house_businesses(search_term)
-    response = make_companies_house_request("search/companies?q=#{search_term}")
-    response["items"].collect do |business|
-      {
-        company_name: business["title"],
-        company_number: business["company_number"]
-      }
-    end
-  end
+  class CompaniesHouseClient
```
Comment (broder) [inline app/helpers/businesses_helper.rb]: This feels a bit weird. Should our `CompaniesHouseClient` be a singleton somewhere? Something like https://stackoverflow.com/questions/29836165/using-one-instance-of-client-for-external-service-throughout-rails-application

File: app/controllers/businesses_controller.rb [hunk 4]
```diff
@@ -33,13 +28,13 @@ def edit; end
 
   # GET /businesses/search_companies_house
   def search_companies_house
-    @businesses = @companies_house_client.companies_house_businesses params[:q]
+    @businesses = companies_house_client.companies_house_businesses params[:q]
```
Comment (broder) [inline app/controllers/businesses_controller.rb]: I think we could just do `CompaniesHouseClient.instance. companies_house_businesses` and then we don't need anything in the helper (unless you think the helper will come in useful with some of the other functionality that you're adding!)

Comment (AnthonyPengelly) [reply app/controllers/businesses_controller.rb]: Yeah, that seems reasonable!

**Pull request merged**